app/clustercoordinators/clustercoordinator.go

```diff
@@ -0,0 +1,58 @@
+//go:generate mockgen -destination=../../mocks/app/clustercoordinators/mock_clustercoordinator.go -package mock_clustercoordinator github.com/rudderlabs/rudder-server/app/clustercoordinators ClusterManagerFactory,ClusterManager
+
+package clustercoordinator
+
+import (
+	"context"
+
+	"github.com/rudderlabs/rudder-server/utils/types/servermode"
+)
+
+var (
+	DefaultClusterManagerFactory ClusterManagerFactory
```

Comment:
Not sure if we do this, I hope there is no race condition. Check this hack which I had to do because tests were failing: https://github.com/rudderlabs/rudder-server/blob/e494d0ee938ca830400b9841f522c2f693ce6c0e/processor/manager_test.go#L138